### PR TITLE
FT0: time offset calibration envs for global run

### PR DESCRIPTION
**Pull request by afurs**
First step for FT0 time offset calibration workflow within global run

### CODE REVIEW

**Review by davidrohr — APPROVED** [review]

**Comment by afurs** [comment]
Temporary on hold, due to anomalous time offset BC dependency

**Comment by chiarazampolli** [comment]
Hello @afurs , 
Can we move this to WIP till it is ready for review?
Chiara

**Comment by afurs** [comment]
Hello @chiarazampolli ,
is it allowed to merge with FT0 calib turned off for a while?

**Comment by davidrohr** [comment]
sure, with disabled by default we can merge it

File: DATA/common/setenv_calib.sh
Comment (davidrohr) [inline DATA/common/setenv_calib.sh:116]: I think for disabling by default, we should set it to =0 here

**Comment by chiarazampolli** [comment]
Sorry for not having seen this before, David already commented.

File: DATA/common/setenv_calib.sh
Comment (afurs) [reply DATA/common/setenv_calib.sh:116]: do you mean `CAN_DO_CALIB_FT0_TIMEOFFSET=0` - this var?

**Comment by afurs** [comment]
Hello @chiarazampolli @davidrohr could you please merge it or shall I change something in code?

**Comment by martenole** [comment]
Hi @afurs can you please remove the WIP from the title of your PR? Then the CI check will start

**Comment by afurs** [comment]
> Hi @afurs can you please remove the WIP from the title of your PR? Then the CI check will start

Hi @martenole , done

**Comment by afurs** [comment]
Hi @martenole I guess CICD issue is related only to compilation problems

File: DATA/common/setenv_calib.sh
Comment (martenole) [inline DATA/common/setenv_calib.sh:27]: Hi @afurs the check actually fails because you commented out this line here:
```
/sw/slc7_x86-64/O2DPG/707-slc7_x86-64-local1/DATA/common/setenv_calib.sh: line 122: CAN_DO_CALIB_FT0_TIMEOFFSET: unbound variable
```
`CAN_DO_CALIB_FT0_TIMEOFFSET` must be defined. Can you uncomment it please? And then, if you want to disable the calibration always you should hard-code
`CALIB_FT0_TIMEOFFSET=0`  somewhere around line 143

Comment (afurs) [reply DATA/common/setenv_calib.sh:27]: Ah, sorry, my bad... fixing now

**Comment by afurs** [comment]
@martenole btw, I would like to always suppress calib for synth, tech and cosmic runs, shall I just ask RC for validation of AliECS config(to exclude FT0 from calibration)? or it can be suppressed here via special vars(I saw e.g. `COSMIC`, etc)?

**Comment by martenole** [comment]
By default the calibrations defined in setenv_calib.sh do run in all modes except cosmics, where they are disabled by default. In synthetics and technicals the calibrations are running, but they do not write any output and push nothing to CCDB. Should that not also work for FT0? The advantage is that the processing is always tested, also in our CI and FST checks.

**Comment by afurs** [comment]
If `ccdb-populator` doesn't write anything for synth and tech - then it is okay :)

**Comment by martenole** [comment]
Great. Yes, this is all done automatically here:
https://github.com/AliceO2Group/AliceO2/blob/a5d4c3dc7a3a21cb14b9c3fcbab4f60b4e6fcdb4/prodtests/full-system-test/aggregator-workflow.sh#L20-L48
Your PR looks good then. I'll merge once the CI passes

File: DATA/common/setenv_calib.sh
Comment (martenole) [inline DATA/common/setenv_calib.sh:125]:
```suggestion
( [[ -z ${CALIB_FT0_TIMEOFFSET:-} ]] || [[ $CAN_DO_CALIB_FT0_TIMEOFFSET == 0 ]] ) && CALIB_FT0_TIMEOFFSET=0
```

**Review by martenole — APPROVED** [review]

**Comment by martenole** [comment]
Hi @afurs 
when I try to create a global workflow with this I get some errors on the aggregator:
```
[ERROR] invalid workflow in o2-calibration-ft0-time-offset-calib: No matching output found for FT0/TIME_SPECTRA/0 as requested by data processor "ft0-time-offset-calib". Candidates:
[ERROR] -FT0/CALIB_INFO/0
[ERROR] -CLP/FT0_TIME_CALIB
[ERROR] -CLW/FT0_TIME_CALIB
[ERROR] 
```
And also for the reconstruction workflow on the EPNs:
```
[ERROR] invalid workflow in o2-dpl-output-proxy: No matching output found for <matcher query: (and origin:FT0 (and description:CALIB_INFO (and subSpec:0 (just startTime:$0 ))))> as requested by data processor "calib-output-proxy-fwd_tf". Candidates:
[ERROR] -FT0/RAWDATA
[ERROR] -FT0/CTFDICT/0
[ERROR] -FT0/LookupTable/0
[ERROR] -FT0/TimeSpectraInfo/0
[ERROR] -FT0/DIGITSBC/0
[ERROR] -FT0/DIGITSCH/0
[ERROR] -FT0/RECCHDATA/0
[ERROR] -FT0/RECPOINTS/0
[ERROR] -FT0/TIME_SPECTRA/0
[ERROR] -FT0/CTFDATA/0
``` 
Could you check please?

**Comment by afurs** [comment]
Hi @martenole , as I can see, aggregator waits for `FT0/TIME_SPECTRA/0` which is produced in the first part of workflow(reco). And `o2-dpl-output-proxy` of first part looks for the result of the aggregator `FT0/CALIB_INFO/0` I guess that specified DPL channel in `o2-dpl-output-proxy` is incorrect, let me check script...

**Comment by afurs** [comment]
@martenole could you please check with this PR https://github.com/AliceO2Group/O2DPG/pull/1022 ?

**Comment by afurs** [comment]
@martenole could you try please with this PR https://github.com/AliceO2Group/O2DPG/pull/1022 ?

**Comment by martenole** [comment]
Checking, just need to recompile some things

**Pull request merged**
